impl/src/planning/subplanners.rs

```diff
@@ -439,225 +426,208 @@ impl<'planner> CrateSubplanner<'planner> {
+            dependency_name_opt: Some(pkg.name.to_string()),
+            message: format!(
+              "Unhandlable dependency type {:?} on {} detected! {}",
+              kind, &pkg.name, PLEASE_FILE_A_BUG
+            ),
+          }
+          .into(),
+        )
+      }
     };
 
-    let mut targeted_dep_names: HashMap<String, DependencyNames> = HashMap::new();
+    if self.is_renamed(pkg) {
+      let dep_alias = DependencyAlias {
+        target: self.buildable_target_for_dep(pkg)?,
+        alias: name.replace("-", "_"),
+      };
 
-    let package = self.crate_catalog_entry.package();
-    for dep in &package.dependencies {
-      // This shadow allow for dependencies with target restrictions to override where
-      // to write data about itself.
-      let mut dep_names = &mut default_dep_names;
+      if !dep_set.aliased_dependencies.insert(dep_alias) {
+        return Err(
+          RazeError::Planning {
+            dependency_name_opt: Some(pkg.name.to_string()),
+            message: format!("Duplicated renamed package {}", name),
+          }
+          .into(),
+        );
+      }
+    }
+
+    Ok(())
+  }
 
-      if dep.target.is_some() {
-        // UNWRAP: Safe from above check
-        let target_str = format!("{}", dep.target.as_ref().unwrap());
+  /// Test to see if a dep has been renamed
+  ///
+  /// Currently cargo-metadata provides rename detection in a few places, we take the names
+  /// from the resolution of a package
+  fn is_renamed(&self, dep_package: &Package) -> bool {
+    self
+      .crate_catalog_entry
+      .package()
+      .dependencies
+      .iter()
+      .filter(|x| x.name == dep_package.name)
+      .filter(|x| x.req.matches(&dep_package.version))
+      .filter(|x| x.source == dep_package.source.as_ref().map(|src| src.to_string()))
+      .find(|x| x.rename.is_some())
+      .map_or(false, |x| x.rename.is_some())
+  }
 
-        // Legacy behavior
-        if let Some(platform_details) = &self.platform_details {
-          if let Some(settings_target) = &self.settings.target {
-            let platform = Platform::from_str(&target_str)?;
+  fn is_dep_targetted(&self, target: Option<&String>) -> bool {
+    target
+      .map(|platform| {
+        let potential_targets = self
+          .platform_details
+          .as_ref()
+          .zip(self.settings.target.as_ref());
 
+        match potential_targets {
+          // Legacy behavior
+          Some((platform_details, settings_target)) => {
             // Skip this dep if it doesn't match our platform attributes
-            if !platform.matches(settings_target, platform_details.attrs().as_ref()) {
-              continue;
-            }
+            // UNWRAP: It is reasonable to assume cargo is not giving us odd platform strings
+            let platform = Platform::from_str(platform).unwrap();
+            platform.matches(settings_target, platform_details.attrs())
           }
+          // TODO (GregBowyer): Make this an enum
```

Comment:
Do you think it'd be a lot of work to make this enum? It sounds like a far better solution and I don't think this is used too widely. Can punt but I think it'd be some great polish.